chainerx_cc/chainerx/routines/connection.cc

```diff
@@ -393,6 +394,28 @@ Array Linear(const Array& x, const Array& w, const absl::optional<Array>& b, uin
     return out_matrix.Reshape(out_shape);
 }
 
+Array EmbedId(const Array& x, const Array& w, absl::optional<int64_t> ignore_label) {
+    if (ignore_label.has_value() && (*ignore_label < 0 || *ignore_label >= w.shape()[0])) {
+        throw IndexError{"ignore_label should be greater than equal to 0 and less than" + w.shape()[0]};
+    }
+    if (ignore_label.has_value()) {
+        Array ignore_label_array = Full(x.shape(), *ignore_label, x.dtype(), x.device());
+        Array mask = Where(x == ignore_label_array, -1, x);
+        std::vector<int64_t> m_shape_vec;
+        for (int64_t i = 0; i < mask.ndim(); i++) {
+            m_shape_vec.emplace_back(mask.shape()[i]);
+        }
+        m_shape_vec.emplace_back(1);
+        Shape m_shape{m_shape_vec};
```

Comment:
`Shape` can be used as a vector. You can just write like:

```c++
Shape m_shape = mask.shape();
m_shape.emplace_back(1);
```

Reply:
Done